dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/Encoding.java

```diff
@@ -57,11 +58,15 @@
 
         /**
          * Reads a serialized type-{@link T} object representation from the given input stream and returns the
-         * corresponding object. Implementations should read the entire input stream, but must not close it.
-         * Format-related deserialization errors surface as {@link IllegalArgumentException}. Inputs and outputs
+         * corresponding object.
+         *
+         * <p>Implementations must close the {@link InputStream} and the {@code Closeable response}, except in the
+         * case that an InputStream is returned to the user, e.g. for a binary response type.
+         *
+         * <p>Format-related deserialization errors surface as {@link IllegalArgumentException}. Inputs and outputs
          * must never be null.
          */
-        T deserialize(InputStream input);
+        T deserialize(InputStream input, Closeable response);
```

Comment:
I think we can safely back this out, if we'd like to ensure the response is closed I think we should pass the response object through instead of the stream, but that expands the API surface in a way I'd prefer to avoid.

Reply:
Hang on a sec - there's a bunch of interesting behaviour that we've put into the `Response#close` method, e.g. your LeakDetectingResponse disarms the leak detector.  If we say InputStreams can be closed even if we never call `Response#close`, won't this leak detector complain?

Reply:
I think we have a test for precisely this case :-)
The leak detector shouldn't complain.

Reply:
Ok cool.